ilastik/shell/gui/licenseDialog.py

```diff
@@ -20,98 +20,119 @@
-    LICENSE_SHORT = """<p>Copyright © 2011-2019, the ilastik developers <team@ilastik.org></p>
-<p>
-This program is free software; you can redistribute it and/or
-modify it under the terms of the GNU General Public License
-as published by the Free Software Foundation; either version 2
-of the License, or (at your option) any later version.
-</p>
-<p>
-In addition, as a special exception, the copyright holders of
-ilastik give you permission to combine ilastik with applets,
-workflows and plugins which are not covered under the GNU
-General Public License.
-</p>
-<p>See the <a href="http://ilastik.org/license.html">license page on the ilastik web site</a> for details.</p>
-"""
-    LICENSE_ERROR = """Cannot find LICENSE file in ilastik sources.
-
-Please report this problem to ilastik-devel@ilastik.org.
-
-License information is available on the the ilastik web site at http://ilastik.org/license.html.
-"""
+
+class LicenseDialog(QDialog):
+    LICENSE_SHORT = dedent(
+        """
+        <p>Copyright © 2011-2022, the ilastik developers <team@ilastik.org></p>
+        <p>
+        This program is free software; you can redistribute it and/or
+        modify it under the terms of the GNU General Public License
+        as published by the Free Software Foundation; either version 2
+        of the License, or (at your option) any later version.
+        </p>
+        <p>
+        In addition, as a special exception, the copyright holders of
+        ilastik give you permission to combine ilastik with applets,
+        workflows and plugins which are not covered under the GNU
+        General Public License.
+        </p>
+        <p>See the <a href="https://ilastik.org/license.html">license page on the ilastik web site</a> for details.</p>
+        """
+    )
+    LICENSE_ERROR = dedent(
+        """
+        Cannot find LICENSE file in ilastik sources.
+
+        Please report this problem to team@ilastik.org.
+
+        License information is available on the the ilastik web site at https://ilastik.org/license.html.
+        """
+    )
+    LICENSE_3RD_PARTY_ERROR = dedent(
+        """
+        Cannot find third party license file. This is expected when running ilastik from source in conda.
+
+        If you are seeing this in a release binary, please report this problem to team@ilastik.org.
+        """
+    )
 
     def __init__(self, parent=None):
-        super(LicenseDialog, self).__init__(parent=parent)
+        super().__init__(parent=parent)
 
-        self.setText(self.LICENSE_SHORT)
+        ilastik_path = Path(ilastik.__file__).parent.parent
 
-        global ilastik_path
-        logo_path = os.path.join(ilastik_path, "shell", "gui", "ilastik-logo-alternate-colors.png")
+        # environment variable set in release binary
+        license_path = Path(os.environ.get("ILASTIK_LICENSE_PATH", ilastik_path / "LICENSE"))
+        license_3rd_party_path = Path(os.environ.get("ILASTIK_LICENSE_3RD_PARTY_PATH", "NOTFOUND"))
```

Comment:
the license info is generated (in fact into a single 30k+ lines textfile) during packaging the binary - different OSs and configurations result in different third party packages being included. I searched around the internet a bit in order to see how other people do it and it seems pretty common to keep the main license and third-party stuff separate. But maybe my search was biased towards this kind of arrangement. While IANAL, to me it seems to be the way.

Generating some additional package that has all the licenses is more complicated than having _two_ env variables when in release mode, imo.

Given that the 3rd party bit is 30k+ lines long, I'd at least want people to have the ability to search - hence I thought it would make sense to open information in a browser (or... default program - that behavior is a bit undefined for local files).